server/hotrod/src/main/scala/org/infinispan/server/hotrod/Decoder2x.scala

```diff
@@ -503,6 +504,19 @@ object Decoder2x extends AbstractVersionedDecoder with ServerConstants with Log
       stats += ("removeMisses" -> cacheStats.getRemoveMisses.toString)
       stats += ("totalBytesRead" -> t.getTotalBytesRead)
       stats += ("totalBytesWritten" -> t.getTotalBytesWritten)
+
+      if (!Constants.isVersionPre24(h.version)) {
+         Option(cache.getComponentRegistry.getComponent(classOf[ClusterCacheStats])).foreach(clusterCacheStats => {
```

Comment:
I think you might have to use a `SecurityAction` here to get the component registry. 
If not, the `HotRodServer.knownCacheRegistries` map is pretty useless...